commands/validators.js

```diff
@@ -0,0 +1,118 @@
+const exitOnError = require('../utils/exit-on-error');
+const connect = require('../utils/connect');
+const inspectResponse = require('../utils/inspect-response');
+const bs58 = require('bs58');
+const { utils } = require('near-api-js');
+const BN = require('bn.js');
+const AsciiTable = require('ascii-table');
+
+function findSeatPrice(validators, numSeats) {
+    let stakes = validators.map((v) => new BN(v.stake, 10)).sort((a, b) => a.cmp(b));
+    let num = new BN(numSeats);
+    let stakesSum = stakes.reduce((a, b) => a.add(b));
+    // assert stakesSum >= numSeats
+    let left = new BN(1), right = stakesSum.add(new BN(1));
+    while (true) {
+        if (left.eq(right.sub(new BN(1)))) {
+            return left;
+        }
+        const mid = left.add(right).div(new BN(2));
+        let found = false;
+        let currentSum = new BN(0);
+        for (let i = 0; i < stakes.length; ++i) {
+            currentSum = currentSum.add(stakes[i].div(mid));
+            if (currentSum.gte(num)) {
+                left = mid;
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            right = mid;
+        }
+    }
+}
+
+function diffValidators(currentValidators, nextValidators) {
+    const result = { newValidators: [], removedValidators: [], changeValidators: [] };
+    const validatorsMap = new Map();
+    const nextValidatorsMap = new Map();
+    nextValidators.map((validator) => nextValidatorsMap[validator.account_id] = validator);
+    currentValidators.map((validator) => { 
+        if (!(validator.account_id in nextValidatorsMap)) {
+            result.removedValidators.push(validator);
```

Comment:
why use `.map` and then `.push` results somewhere? This is very confusing, makes sense to use `for (const validator of currentValidators)` or `.forEach` at least.